### PR TITLE
Expanding Conda recipes (w/ and w/o framework) + CD

### LINKED ISSUE #38: Add pure-python conda package
Need a pure-python wrapper package so ADDIE can import this with `mantid-workbench` conda package and not have conflicting versions of Mantid framework in stack

### PR DESCRIPTION
**Pull request by marshallmcdonnell**
This adds recipes for both the w/ framework (`mantid-total-scattering) and w/o framework (`mantid-total-scattering-python-wrapper`) conda recipes. 

Also, the Travis CI is updated to allow for automated deployment of the packages with tagged commits on `master`. 

Added `installation` section for conda in README. **Refer to these instructions for testing**

Finally, there is import testing in the recipes to make sure packages work.

This will close #38, #39, and #40 

Conda packages:
https://anaconda.org/marshallmcdonnell/mantid-total-scattering
https://anaconda.org/marshallmcdonnell/mantid-total-scattering-python-wrapper

To verify that deployment works, see this build:
https://travis-ci.org/marshallmcdonnell/mantid_total_scattering/builds/529856458

To verify deployment will not happen on non-tagged commit on non-master branch, see this (latest) build:
https://travis-ci.org/marshallmcdonnell/mantid_total_scattering/builds/529911561

### CODE REVIEW

**Comment by marshallmcdonnell** [comment]
The CI "failure" is for coverage, which can be ignored.

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,80 @@
+
```
Comment (peterfpeterson) [inline .gitignore]: Is this basically [github's gitignore](https://github.com/github/gitignore/blob/master/Python.gitignore)?

File: .travis.yml [hunk 2]
```diff
@@ -5,16 +5,40 @@ services:
...
+  # Docker w/ framework
   - os: linux
-    env: TYPE="conda" CONDA=2.7
+    env: TYPE="docker"
     sudo: required
+
+  # Conda w/o mantid-framework
   - os: linux
-    env: TYPE="conda" CONDA=3.6
+    env: TYPE="conda" CONDA=2.7.14 PKG="mantid-total-scattering-python-wrapper"
```
Comment (peterfpeterson) [inline .travis.yml]: Is it possible to drop the patch level from the version specification like in the 3.6 case?

File: .travis.yml [hunk 3]
```diff
@@ -30,22 +54,26 @@ before_install:
...
-      conda install -n test-env -c mantid mantid-workbench nexus poco;
-      conda install -n test-env flake8;
     fi
 
+
 install:
   - |
     if [[ "${TYPE}" == "conda" ]]; then
-      source activate test-env; 
+      conda config --set always_yes yes --set changeps1 no --set anaconda_upload no;
```
Comment (peterfpeterson) [inline .travis.yml]: `--set always_yes yes` is one of the best command line options I've ever seen.

File: .travis.yml [hunk 4]
```diff
@@ -57,27 +85,29 @@ script:
...
+ # Deploy pure-python package to PyPI https://pypi.org/project/mantid-total-scattering/
+ - provider: pypi
+   user: mcdonnellmt
+   skip_cleanup: true
+   skip_existing: true
+   on:
+     tags: true
+     branch: master
+   password:
+     secure: kqA6ITb6fr+9Q/KBeU/mxa+d319Uwgc9wi8OPsZssvSneQ8vGAIVSSDBeCVbKSXLjXUIKUWYfndOiqT8wCQU+jpY1TZG0Bm0zWGGxBvcUJEnVrB7siWJ6l+MZ4nPXFFMtwhP0KMDtwen0fLnGOLQlNLyz3qb0Lf8NkM1BumXDK7Lc7p/x8ziKHQothL047jx+0PMsydbTdK2gae/ADm6R4z+ZJirv0/yciUQnCwmiXSnYX5UHtkXqP3k21ZtTV3eA9vNvK1HMEti51lFDrgTt5EBOxtX0UIz6YqfCl5ZHyXMS4uCwqL/m7gRPAc1MTmSurZ4JJkvsaVrU3vUdIimlm2EwqH6+yOfjvWlTdwjglkplt4Z95/rN3CLvYIrYq76XsG2yxY8c11+xmpSMo2/6vbRvmKhmbyV6tFeBgq5rflA/xJkM5WqSyMcE0lhk8xc0RL7aAeblyAs9q/hN9BiZIesegE6pd8DWYEGw6cWnwzaflLgze0nJYE5gQ5z5LZJky8/Nxdxumtf9XVS48PenaNIbDLLG1ER2nufHHg3/5Yv+c0TMRWKv6plb4OLj2AIYa2aSedm7YyVhCbVV0/xM0HxZ8fpKxBMDuFuuU1S1YgQkex7cCfj2QW2JBGoX/iAvPz1sZcU7A+VKaKMJyY+SSmQf+Sz3Px3O8CE+9LerV4=
```
Comment (peterfpeterson) [inline .travis.yml]: This should be set in travis through the web config

File: setup.py [hunk 5]
```diff
@@ -13,19 +13,14 @@
       description='Mantid Total Scattering Reduction',
       long_description_content_type="text/markdown",
       license='GPL License (version 3)',
-      scripts=['bin/mantidtotalscattering'],
+      entry_points = {
+        'console_scripts': [
```
Comment (peterfpeterson) [inline setup.py]: This magic should get into addie as well.

Comment (marshallmcdonnell) [reply setup.py]: :+1: Working on it now. Have to move and re-structure the `main` function call (hence how I found out ADDIE is an `Image Changer` :smile:)

File: .travis.yml (hunk 2 above)
Comment (marshallmcdonnell) [reply .travis.yml]: Ah, yes, probably can for the wrapper. We are stuck for the one with the mantid-framework included. Good catch

File: .gitignore (hunk 1 above)
Comment (marshallmcdonnell) [reply .gitignore]: This one is from https://www.gitignore.io/api/pydev,python,eclipse

File: .travis.yml (hunk 4 above)
Comment (marshallmcdonnell) [reply .travis.yml]: Done! :+1:

**Pull request merged**